src/basic/path-util.c

```diff
@@ -639,14 +639,31 @@ static int check_x_access(const char *path, int *ret_fd) {
         return 0;
 }
 
-int find_executable_full(const char *name, bool use_path_envvar, char **ret_filename, int *ret_fd) {
+int find_executable_full(const char *name, const char *root, bool use_path_envvar, char **ret_filename, int *ret_fd) {
         int last_error, r;
         const char *p = NULL;
 
         assert(name);
 
         if (is_path(name)) {
                 _cleanup_close_ int fd = -1;
+                _cleanup_free_ char *path_name = NULL;
+
+                /* Function chase_symlinks is invoked only when root is not NULL,
+                 * as using it regardless of root value would alter the behavior
+                 * of existing callers */
```

Comment:
in which way?

Reply:
If you have eg `/bin/sleep` find_executable right now returns `/bin/sleep`, but with the unconditional change it would start returning `/usr/bin/sleep`. I suggested to do this to avoid unforeseen behavior changes that could introduce regression or make things more complicated.

Reply:
ah, this is a much better explanation, please update the comment with something like this